### PR TITLE
[withdraw]: Refactor sym. fee + tx handling 

### LINKED ISSUE #446: Liquidity: Send withdraw (unstake) transaction
- Show loading state using TxModal
- Handle incoming events from Binance (or other) chain to show result
- Reload balances of un-staked assets

### PR DESCRIPTION
**Pull request by veado**
- [x] Re-write fee handling for WITHDRAW txs
- [x] Add `withdraw$`
- [x] Refactor UI to show fee + tx state
- [x] Update stories
- [x] Update i18n

Closes #446

### CODE REVIEW

File: src/renderer/components/deposit/withdraw/Withdraw.tsx [hunk 1]
```diff
@@ -1,284 +1,339 @@
...
+        target={{ asset, amount: assetAmountToWithdraw }}
+        source={O.some({ asset: AssetRuneNative, amount: runeAmountToWithdraw })}
+        stepDescription={stepDescription}
+      />
+    )
+  }, [intl, asset, withdrawState, assetAmountToWithdraw, runeAmountToWithdraw])
+
+  const onCloseTxModal = useCallback(() => {
+    // unsubscribe
+    unsubscribeWithdrawSub()
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/components/deposit/withdraw/Withdraw.tsx:225]: but `setWithdrawSub` already does it. is not it ?

File: src/renderer/components/deposit/withdraw/Withdraw.tsx [hunk 2]
```diff
@@ -1,284 +1,339 @@
...
+  const [showPasswordModal, setShowPasswordModal] = useState(false)
+
+  const closePasswordModal = useCallback(() => {
+    setShowPasswordModal(false)
+  }, [setShowPasswordModal])
+
+  const onClosePasswordModal = useCallback(() => {
+    // close password modal
+    closePasswordModal()
+  }, [closePasswordModal])
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/components/deposit/withdraw/Withdraw.tsx:306]: seems this are the same methods. is there any sense why we should split it ?

File: src/renderer/i18n/ru/common.ts [hunk 3]
```diff
@@ -54,6 +54,7 @@ const common: CommonMessages = {
   'common.tx.sending': 'Отправка транзакиции',
   'common.tx.sendingAsset': 'Отправка транзакиции {assetSymbol}',
   'common.tx.checkResult': 'Проверка результата',
+  'common.tx.view': 'View {assetSymbol} transaction - RU',
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/i18n/ru/common.ts:57]: ```'common.tx.view': 'Посмотреть {assetSymbol} танзакцию',```

File: src/renderer/i18n/ru/deposit.ts [hunk 4]
```diff
@@ -40,16 +40,17 @@ const deposit: DepositMessages = {
   'deposit.wallet.add': 'Добавить',
   'deposit.wallet.connect': 'Пожалуйсста добавьте свой кошелек',
   'deposit.pool.noDeposit': 'У вас нет доли в этом пуле',
+  'deposit.withdraw.pending': 'Withdraw - RU',
+  'deposit.withdraw.success': 'Successful withdraw - RU',
+  'deposit.withdraw.error': 'Withdraw error - RU',
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/i18n/ru/deposit.ts:45]:
```
'deposit.withdraw.pending': 'Изъять',
'deposit.withdraw.success': 'Успешно узъяли',
'deposit.withdraw.error': 'Ошибка при изъятии',
  ```

File: src/renderer/i18n/ru/deposit.ts [hunk 5]
```diff
@@ -40,16 +40,17 @@ const deposit: DepositMessages = {
...
   'deposit.withdraw.receiveText': 'Вы полчучите',
   'deposit.withdraw.fees': 'Transaction fee: {thorMemo}, Outbounding fees: {thorOut} + {assetOut} - RU',
   'deposit.withdraw.feeNote': 'Важно: {fee} BNB останется на вашем кошельке для покрытия комисий.',
   'deposit.withdraw.drag': 'Перетащите для изъявтия',
-  'deposit.withdraw.add.error.thorMemoFeeNotCovered':
-    'Transaction fee of {fee} needs to be covered by your balance (currently {balance}) - RU',
-  'deposit.withdraw.add.error.outFeeNotCovered':
-    'Outbounding fee of {fee} needs to be covered by receiving amount (currently {amount}) - RU'
+  'deposit.withdraw.error.feeNotCovered':
+    'Transaction fee of {fee} needs to be covered by your balance (currently {balance}) - RU'
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/i18n/ru/deposit.ts:53]:
```
'deposit.withdraw.error.feeNotCovered':
    'Комиссия транзакции {fee} должна покрываться вашим балансом (баланс: {balance})'
```

File: src/renderer/services/chain/transaction/withdraw.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+const { pools: midgardPoolsService, validateNode$ } = midgardService
+
+/**
+ * Symetrical withdraw stream does 3 steps:
+ *
+ * 1. Validate pool address + node
+ * 2. Send withdraw transaction
+ * 3. Check status of both transactions
+ *
+ * @returns WithdrawState$ - Observable state to reflect loading status. It provides all data we do need to display status in `TxModul`
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/services/chain/transaction/withdraw.ts:27]: typo `TxModul` => `TxModal`

**Review by thatStrangeGuyThorchain — DISMISSED** [review]
just some questions

**Comment by thatStrangeGuyThorchain** [comment]
and pls fix tests :)

File: src/renderer/components/deposit/withdraw/Withdraw.tsx (hunk 2 above)
Comment (veado) [reply src/renderer/components/deposit/withdraw/Withdraw.tsx:306]: Good find, just an accident by copy / paste from other sources

File: src/renderer/components/deposit/withdraw/Withdraw.tsx [hunk 7]
```diff
@@ -1,284 +1,339 @@
...
-    const runeBalance = FP.pipe(
-      oRuneBalance,
-      O.map(baseToAsset),
-      O.getOrElse(() => ZERO_ASSET_AMOUNT)
+  useEffect(() => {
+    // Unsubscribe of (possible) previous subscription of `withdraw$`
+    return () => {
+      unsubscribeWithdrawSub()
+    }
+  }, [unsubscribeWithdrawSub])
```
[the remark this answers is not in the document]
Comment (veado) [reply src/renderer/components/deposit/withdraw/Withdraw.tsx:126]: yes

File: src/renderer/components/deposit/withdraw/Withdraw.tsx (hunk 1 above)
Comment (veado) [reply src/renderer/components/deposit/withdraw/Withdraw.tsx:225]: Correct, I will remove that line ^

**Pull request merged**